### PR TITLE
fixed all templates files not being copied

### PR DESCRIPTION
**Pull request by creepinson**
# Pull Request

## Description:

 

I must have messed up the copyfiles command earlier which caused the previous pr to make it not generate all of the required files (only src/ was generated).  Sorry about this lol

Skywolf fixed the generation command for the files as well as removing the extra template/src/index.ts file, and I fixed concurrency problems so that it should generate node_modules after.

## Related Issue:

Closes #XXXXX

## Checklist

- Copyfiles command fixed
- Concurrency issue fixed
- Removed src/template/src/index.ts cuz it doesn't need to be there.

### Semantic Versioning

My pull request updates the version in the `package.json` based on the following:

- [ ] Major Update (X.\_._) - this PR introduces breaking changes.
- [ ] Minor Update (\_.X._) - this PR introduces new, but non-breaking, features.
- [x] Patch Update (\_._.X) - this PR adds no new features (i.e. bug fixes, documentation updates)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   "name": "tsprogen",
-  "version": "1.2.0",
+  "version": "1.2.1",
   "description": "A CLI project generator for setting up TypeScript + Node projects.",
   "main": "./prod/bin.js",
   "bin": {
     "tsprogen": "./bin/tsprogen"
   },
   "scripts": {
     "build": "tsc",
```
Comment (nhcarrigan) [inline package.json:10]:
```suggestion
  "scripts": {
    "prebuild": "rm -rf ./prod",
    "build": "tsc",
```

While we are in here, let's add a prebuild script to remove the existing `prod` directory - this'll prevent conflicts when building and publishing.

**Review by SkyWolf369 — APPROVED** [review]

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
One tweak. 

Additionally, what happened to the `src` directory generation? And the `index.ts` file generation? 🤔

**Review by SkyWolf369 — APPROVED** [review]

**Review by nhcarrigan — APPROVED** [review]

**Comment by creepinson** [comment]
About the src directory, skywolf said we didn't need it. I mean, its kind of true because you can just create a src folder yourself but I guess it would be nice to have that there.

**Comment by nhcarrigan** [comment]
S'okay - I've merged a fix for it.

The `tsconfig.json` generates with a setting to compile from `src` into `prod`, so it's good to generate that folder (and the `index.ts` it contains) as part of the CLI command.

**Pull request merged**